botocore/docs/bcdoc/restdoc.py

```diff
@@ -205,7 +221,25 @@ def flush_structure(self):
                     self.style.link_target_definition(refname, link)
         value = self.getvalue()
         for name, section in self._structure.items():
-            value += section.flush_structure()
+            # Checks is the AWS API Documentation link has been generated.
+            # If it has been generated, it gets passed as a the doc_link parameter.
+            if b'`AWS API Documentation <' in value:
+                start = value.find(b'`AWS API Documentation <')
+                end = value.find(b'>`_', start) + 3
+                docs_link = value[start:end] + b'\n\n'
+            value += section.flush_structure(docs_link)
+
+        # Replace response/request sections if the line number exceeds our limit.
+        # The section is replaced with a message linking to AWS API Documentation.
+        line_count = len(value.splitlines())
+        line_limit = SECTION_LINE_LIMITS.get(self.name)
+        if line_limit and line_count > line_limit:
+            value = (
+                f'\n\n    **{SECTION_NAME_MAPPING.get(self.name)}**'
+                f'\n{LARGE_SECTION_MESSAGE}'
+            ).encode()
```

Comment:
Let's make this a template and we can do something like:

```suggestion
            section_name = SECTION_NAME_MAPPING.get(self.name)
            value = TEMPLATE.format(section_name)
```

There's also a question of can we ever hit this with a section name that doesn't exist in the mapping? Do we just write `None`?

Reply:
Not currently, but it does leave the possibility if `SECTION_LINE_LIMITS` and `SECTION_NAME_MAPPING` don't have exactly matching keys for whatever reason. Maybe combining those into one dictionary would be a bit better. Something like:

```python
line_count = len(value.splitlines())
line_limit_config = SECTION_LINE_LIMIT_CONFIG.get(self.name)
if line_limit_config and line_count > line_limit_config['limit']:
    value = (
        f'\n\n.   **{line_limit_config['name']}**\n{LARGE_SECTION_MESSAGE}'
)